### DISCUSSION TITLE
How to handle Tag onClose() event and get the item that was closed in TagPicker component?

### DISCUSSION THREAD

**Discussion by ivangabriele** [rsuite/rsuite#3213] [Q&A] [answered]
I have a case with a lot of `TagPicker` customization and handle changes with `onClean` and `onSelect` props (rather than using `onChange`).  I use these props rather than `onChange` because they give me the complete item rather than just the value which avoids a lot of re-mapping back in my case.

But I wonder how I can handle a Tag close.

There is a `tagProps` prop to which I can pass `{ onClose: handleTagClose }` but I can't do much with that since its only param `event` seems to be a React Synthetic Event.

I wonder how I could know which data item tag has been closed when it happens?

**Comment by SevenOutman** [comment, marked as answer]
Hi @ivangabriele, I've seen user [using `renderValue`](https://github.com/rsuite/rsuite/issues/2402) to render those Tags, and therefore bind `onClose` on each Tag. Would that help?

**Reply by ivangabriele** [reply]
Oh indeed! That's a bit verbose but that's a perfectly workable solution. Thank you :).

**Reply by SevenOutman** [reply]
That said, maybe some new prop like `onRemoveTag(value, item)` would be useful. I'd add it to the backlog. Thanks for raising this question :)
